nitro-protocol/contracts/libraries/NitroUtils.sol

```diff
@@ -0,0 +1,238 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.7.6;
+pragma experimental ABIEncoderV2;
+
+import '../interfaces/INitroTypes.sol';
+
+library NitroUtils {
+    // *****************
+    // Signature methods:
+    // *****************
+
+    /**
+     * @notice Require supplied stateHash is signed by signer.
+     * @dev Require supplied stateHash is signed by signer.
+     * @param stateHash State hash to check.
+     * @param sig Signed state signature.
+     * @param signer Address which must have signed the state.
+     * @return true if signer with sig has signed stateHash.
+     */
+    function isSignedBy(
+        bytes32 stateHash,
+        INitroTypes.Signature memory sig,
+        address signer
+    ) internal pure returns (bool) {
+        return signer == NitroUtils.recoverSigner(stateHash, sig);
+    }
+
+    /**
+     * @notice Check if supplied participantIndex bit is set to 1 in signedBy bit mask.
+     * @dev Check if supplied partitipationIndex bit is set to 1 in signedBy bit mask.
+     * @param signedBy Bit mask field to check.
+     * @param participantIndex Bit to check.
+     * @return true if supplied partitipationIndex bit is set to 1 in signedBy bit mask.
+     */
+    function isClaimedSignedBy(uint256 signedBy, uint8 participantIndex) internal pure returns (bool) {
+        return ((signedBy >> participantIndex) % 2 == 1);
+    }
+
+    /**
+     * @notice Check if supplied participantIndex is the only bit set to 1 in signedBy bit mask.
+     * @dev Check if supplied participantIndex is the only bit set to 1 in signedBy bit mask.
+     * @param signedBy Bit mask field to check.
+     * @param participantIndex Bit to check.
+     * @return true if supplied partitipationIndex bit is the only bit set to 1 in signedBy bit mask.
+     */
+    function isClaimedSignedOnlyBy(uint256 signedBy, uint8 participantIndex) internal pure returns (bool) {
```

Comment:
Indeed, we could have used `isClaimedSignedBy` instead, but that would loosen the restriction of 'strict turntaking'. I think, it would be nice to have such restriction, in case DApp developers make some mistake and provide correct amount of signatures, but incorrect claiming. This could point on some mistakes in their code.
Regarding gas consumption: it must be nearly the same as bit shift and calculating the power of 2 end up taking roughly the same number of bit operations.
That being said, I believe it is nice we can point to a potential mistake in DApp usage without paying more (or even paying the same) gas!